### ISSUE TITLE
Join on IS NOT FALSE clause produces wrong results.

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#2130]
Test case:

```
create table booltest (b bool);
insert into booltest values ('t');
insert into booltest values (null);
select * from booltest a, booltest b where (a.b = b.b) is not false;
 b | b 
---+---
 t | t
(1 row)
```

That result is incorrect. `x IS NOT FALSE` is true, when `x IS NULL`. The correct result would be:
```
 b | b 
---+---
 t | t
 t | 
   | t
   | 
(4 rows)
```

Spotted this while reading code. It's caused by this code that was added into `make_restrictinfo_internal` function in GPDB:
```

	/**
	 * If this is a IS NOT FALSE boolean test, we can peek underneath.
	 */
	if (IsA(clause, BooleanTest))
	{
		BooleanTest *bt = (BooleanTest *) clause;

		if (bt->booltesttype == IS_NOT_FALSE)
		{
			clause = bt->arg;
		}
	}
```
That is wrong, it assumes that "x IS NOT FALSE" is equal to just "x", but that is wrong, because they handle NULLs differently.

**Comment by vraghavan78** [comment]
```
vraghavan=# select * from booltest a, booltest b where (a.b = b.b) is not false;
 b | b
---+---
   | t
 t | t
   |
 t |
(4 rows)

vraghavan=# set optimizer  = off;
SET
vraghavan=# select * from booltest a, booltest b where (a.b = b.b) is not false;
 b | b
---+---
 t | t
(1 row)
```
Planner issue. Tagged as such. Thanks @hlinnaka

**Comment by hlinnaka** [comment]
Fixed, in commit 60787a9ecd.
